build.gradle

```diff
@@ -344,33 +335,33 @@ project(':iceberg-core') {
     api project(':iceberg-api')
     implementation project(':iceberg-common')
     implementation project(path: ':iceberg-bundled-guava', configuration: 'shadow')
-    annotationProcessor "org.immutables:value"
-    compileOnly "org.immutables:value"
+    annotationProcessor libs.immutables.value
+    compileOnly libs.immutables.value
 
-    implementation("org.apache.avro:avro") {
+    implementation(libs.avro.avro) {
       exclude group: 'org.tukaani' // xz compression is not supported
     }
 
-    implementation 'io.airlift:aircompressor'
-    implementation 'org.apache.httpcomponents.client5:httpclient5'
-    implementation "com.fasterxml.jackson.core:jackson-databind"
+    implementation libs.aircompressor
+    implementation libs.httpcomponents.httpclient5
+    implementation platform(libs.jackson.bom)
```

Comment:
I think we should have a `jackson-databind = { module = "com.fasterxml.jackson.core:jackson-databind", version.ref = "jackson-ver" }` definition and then this would be using `implementation libs.jackson.databind`

Reply:
This approach using bom was suggested by @snazy in https://github.com/apache/iceberg/pull/7694#discussion_r1227938340 & https://github.com/apache/iceberg/pull/7694#discussion_r1227943762
which actually has its own advantages

Reply:
Added it back, I think I removed it by accident, thanks!
Though all the check still passed somehow even without it 🤷